### DISCUSSION TITLE
Combined auto crafting in an advanced block

### DISCUSSION THREAD

**Discussion by maxtraxv3** [RS485/LogisticsPipes#1593] [Ideas]
is there a possablity you could make an Advance crafing block(s) so you dont end up with 2000+ auto crafting blocks?

i love logistics, but AE(1&2) has a better crafting system any chance you could improve yours just a bit? like say a patten storage, and like a block that needs BC lasers to craft?

**Comment by hron84** [comment]
LP will eventually provide proper integration with AE2 (it is already on the roadmap) so I don't think they want to work on this issue. LP has a long and very deep friendship with AE2, the LP crafting itself is just a one step on autocrafting journey, it is quite cheap to start, and satisfying if you do not play with tech-intensive mods.

**Comment by Nex4rius** [comment]
But not everyone is using lp and ae2 together so its still a good idea to make logisticspipes crafting more compact to not have hundreds of crafting tables (it would also increase the perfomance if you dont have to spam so many crafting tables).

Maybe a crafting table that works with the chassis pipe and 8 crafting modules ?

**Comment by maxtraxv3** [comment]
ah man, i hope not, i dislike AE2(i liked AE 1), i hope they would find a better standalone thing, and there about the same in resource costs, the definitely not cheaper.

**Comment by bziemons** [comment]
@hron84 AE2 still has no friendship with LP ;)
Other than that I do look at LP as a distributed system, where you do not have one block which does everything.

**Comment by Nex4rius** [comment]
Btw the problem without the proper integration in ae2 is known since 15.3.2014 ... 16 months ... so I dont think we will see this at all in ae2 http://ae-mod.info/Tracker/0192/

We dont want one block that does everything just a little bit more compact because if you are playing with gregtech there are tons of steps involved in making machines.

**Comment by maxtraxv3** [comment]
i agree with above, i'm most certainly not asking for a block that does every thing, just a better was about it.

AE2 autocrafting system(apart from the compactstorage) is the only reason any one uses AE2

**Comment by hron84** [comment]
@davboecki we can still have the distributed system itself. For example, we can have an upgrade on crafting pipes that allow them to contain more than one recipe - for example one more on each tier. Since our tier system is quite limited, we can upgrade it to 3 recipe or so? It is still not too compact, but makes you require 66% less autocrafting table. The balancing point is obviously the upgrade itself. 

The hardest part is the autocrafting table behind these pipes. Currently, we have a wooden table, that remembers to one recipe, and we have to set it before we can import it to the pipe. The current table is not applicable to this advanced autocrafting, since it is a wooden table infused with wood and stone and they're a very basic materials, not able to remember more than one recipe. But, what happens if we infuse the basic autocrafting table with some more advanced materials, for example metals or gems and redstone? Maybe it can expand the crafting memory? I dunno.

**Comment by QORTEC** [comment]
What I was thinking; maybe make new tier of the beloved Logistics Crafting Table and add more complex mechanics to the next tier.

What would happen if you were to totally separating the crafting tables form it's recipes?

What if you were to store all your crafting patterns in a Logistics Pattern Provider, and then send the pattern upon request to the Crafting Table. And after the table is done crafting the requested items it would send the pattern back to the originating Pattern Provider.

Now by adding limitations to this idea we can make things a bit more interesting; A low tier limitation could be that you need to “pair” a single Crafting Table to a single Pattern Provider, this would prevent a player from using a single Crafting Table for a complicated multi-step item.

Though if we want to add a 3rd tier we would probably want to slightly negate the 2nd tier limitations. To negate the limitation we could add a Logistics Crafting Controller, this block would allow you to connect a number of Pattern Providers and Crafting Tables to a single Crafting Controller. By doing so it would allow multiple crafting tables to access the Patterns in multiple Pattern Providers, this would allow a more compressed build but still not use a single magic block that does everything.

Also as long as the Pattern Provider doesn't have a large storage capacity we will always be far away form the single magic block wonder.

**Edits:**
15-07-22: corrected spelling

**Comment by ruifung** [comment]
Well, that sounds somewhat like the crafting system in AE2 right now, which stores patterns in a ME interface adjacent to the Assembler. And uses "Crafting CPU's" that act as some kind of buffer for the crafting materials.

Also its tier, not tear lol. I think you mistyped 3rd as 3ed too.

**Comment by QORTEC** [comment]
@ruifung your right, after looking at the AE2 crafting and the crafting system I proposed has similarities to AE2. I should probably play around with Applied Energistics before I make any comments about their crafting system.

I was originally under the [false] impression that AE had a crafting system that could learn an “infinite” amount of patterns [stored across a parallel pattern storage system], and also all of those patterns would be available for crafting at any time. But according to my current research this doesn't seam to be the case.

**Comment by bziemons** [comment]
@maxtraxv3 could you please leave the title when I change it to better represent the contents of this issue? I exaggerated when I said "magic block that does everything" in my comment, but this title just says "combined" and that is what you want. I basically reused your wording. If something is still wrong, please say a word and do not change it back.

@hron84 I have no idea why you repeated to davboecki with the distributed part, but if I may answer on your comment: I know that combining a few blocks in one will not destroy Logistics Pipes in it's spirit ;) I do see a problem with crafting pipes and the way they function and import recipes.

**Comment by hron84** [comment]
@theZorro266 You misunderstood me. The way how crafting pipes working is awesome as I think, I do not want to see any change in the current basic behavior. I would like to see crafting pipes can import more than one recipe from multiple crafting tables or from one specific crafting table that can hold same or less recipe than the pipe can. It would solve the opening problem (smaller systems), and do not break the current behavior how crafting pipes currently work. Also it would still require same or more resource to build the system. It can avoid to introduce something like AE2 crafting patterns (we currently have a more awesome solution for it) or mimicing how the AE2 system is works, since I think it would not fit to... ahem... to LP's spirit. :smile: 

Anyway, you made an awesome stuff, and I like every single block/item in this mod, so don't get me wrong, I wouldn't like to change the world. Just sharing ideas. I can really accept if you think my idea is on wrong way or you have totally different plans that are currently not public.

**Comment by bziemons** [comment]
@hron84 right, the crafting pipe needs to be changed and the possibility to chose or import multiple recipes sounds hard to me, when I think of the code around that functionality. Other than that I have no secret plans, because I am not afraid of discussions. My actual opinion on this is, that I don't need this feature.

**Comment by maxtraxv3** [comment]
@theZorro266  so your saying the 3 blocks it would take to make this is one block that does everything? (storage, autocrafting table, lazers) no to mention the power req, or the pipes you would need to do this is considered one? i dont get it why would you not wanna break away from AE rather then basically melt into a background where less and less people use this mod???

**Comment by bziemons** [comment]
I do not really understand your first two questions, but I got a clue on your third "question". I do not see a reason to "break away" from AE, because I do not see LP be a part of AE nor do I see LP being similar to AE. Basically LP right now is very much similar to the LP it always was and you can definitely use it the same way. Over the years many new features were added to Logistics Pipes and it evolved and performs much better than it's historical versions.

Now to the point: I do not see a reason to make LP easier or more comforting so everyone wants to use the mod. There always were people and there are people that do not like to care for their logistics and instead just go with it. And I am pretty much sure that this mod never was intended for such people, hence the word "Logistics" in it's name. I do like the mod as it is, but I do not mean to say that I do not want to see any more evolving or extensions on it. In fact there are already quite a few new features planned for 0.10 and already being worked on. And in the end I do not care for the number of players either. It is not like this is a commercial product and therefor a "sinking ship".

**Comment by maxtraxv3** [comment]
easier ? try more "functional" cost for it could be sky high, far as i care.

**Comment by Sakata-MC** [comment]
As a long-time user of LP, I agree with Zorro and Dav completely.  While I would like to see a crafting pipe (reduces %50 of the 'blocks'), I love and understand LP. I have done and still do some really crazy things with LP that push the envelope on what Zorro and crew had in mind when they designed the mod.

Being an old school programmer myself, I also realize that what you're requesting would pretty much require a HUGE overhaul of the crafting system's code, if not a complete re-write. Having multiple recipes on one block or pipe would be more work than I think it'd be worth.

Also, LP can still stomp AE2 into the ground in terms of crafting speed when it comes down to complex item crafting that requires many steps.  ESPECIALLY if that item is requested in any usable amount.  LP can craft the AE 64K cells faster than AE can :P  I will however say that there are some cases where I find AE fits the bill for certain crafting recipes or edge cases, but in my case I always have these feed back into LP anyways.

Plus, on the aesthetic side, I LIKE seeing items flying across my crafting areas while LP is crafting.  It is great that it looks like an assembly line where the components for complex items are made in stages and the final product gets back to me.

**Comment by bziemons** [comment]
@Sakata-MC _cough_

> Having multiple recipes on one block or pipe would be more work than I think it'd be worth.

To give you an update: There are crafting modules now that do exactly that. In the end this change would still require a lot of work though, so you are not wrong.

**Comment by 1n5aN1aC** [comment]
Since this issue is tagged as "Needs comments," I thought I'd leave my 4 cents here....

While I am somewhat against a AE-like multi-block crafter with many recipes, what I've wanted for a long time, and could be a decent compromise, has been a combined Crafting pipe and Crafting Table Block.

So the basic idea may have already been proposed before, but consists of a single block, resembling a Crafting table, but with 2 primary changes:
1.  It acts as a crafting logistics pipe, just without the extra features those have, such as satellite pipes, etc.
2.  Preferably, it acts like request tables, and also has a built-in pipe, so many can be stringed together, and items routed internally.

**Comment by loordgek** [comment]
@1n5aN1aC i like to see the items flowing through the pipes
